src/main/kotlin/exercise4/task2/TransactionalBankAccount.kt

```diff
@@ -78,30 +80,144 @@ import java.util.*
  *      ```
  */
 
-
-
-
 private val currentTime: LocalDateTime get() = LocalDateTime.now()
 
 private fun LocalDateTime.prettyPrint(): String {
     val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm")
     return this.format(formatter)
 }
 
+enum class TransactionType {
+    DEPOSIT, WITHDRAWAL
+}
+
+enum class TransactionStatus {
+    SUCCESS, FAILURE
+}
+
+data class Transaction(
+    val transactionDate: LocalDateTime,
+    val transactionType: TransactionType,
+    val amount: Double,
+    val oldBalance: Double,
+    val newBalance: Double,
+    val transactionStatus: TransactionStatus
+)
+
+class TransactionalBankAccount(
+    private val accountNumber: String,
+    private val accountHolderName: String,
+    private var balance: Double = 0.0
+) : BankAccount(accountNumber, accountHolderName, balance) {
+
+    private var transactions = mutableListOf<Transaction>()
```

Comment:
Property should be defined as a read-only using `val` keyword, since we do not modify reference to the list after we initialized it.